src/templates.rs

```diff
@@ -360,22 +360,23 @@ pub static TEMPLATE_QUERY_PARAM_SCRIPT: &str = r#"
         if (queryParams.size === 0) return url; // No query params, return original URL
 
         function appendQueryParams(url) {
-            const parsedUrl = url;
-            const noOrigParams = parsedUrl.lastIndexOf('?') == -1;
+            const hashIndex = url.lastIndexOf('#');
```

Comment:
Naw man. JS has a dedicated URL class. Use it. https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

Reply:
Using it requires us to know the `<base>` of the target page, which requires us to load it, which requires this anyway. The URL class doesn't support working with raw strings that are supposed to be relative URLs.

Reply:
Fixed in 474e4a2.